src/helpers/common.ts

```diff
@@ -14,3 +16,16 @@ export async function getAccountData<T>(
 
   return (await program.account[accountType].fetchNullable(address)) as T;
 }
+
+/**
+ * Convert amount to lamports and return as a BN
+ */
+export function convertToLamportsBN(
+  amount: number | string,
+  tokenDecimal: number
+): BN {
+  const valueInLamports = new Decimal(amount).mul(
+    Decimal.pow(10, tokenDecimal)
+  );
+  return new BN(valueInLamports.toString());
```

Comment:
it should be: `valueInLamports.floor().toString()`
It will get error in case `valueInLamports` is odd number